### PR TITLE
Python client changes for transient MRF feature. isMRF flag is added.

### PR DESCRIPTION
**Pull request by anshulmittal-flex**

### CODE REVIEW

File: tests/simulation/params/test_validators_params.py [hunk 1]
```diff
@@ -610,3 +610,72 @@ def test_rotation_parent_volumes():
...
+    c_2 = Cylinder(
+        name="outer_rotating_cylinder",
+        outer_radius=12 * u.cm,
+        height=12 * u.cm,
+        center=(0, 0, 0) * u.cm,
+        axis=(0, 0, 1),
+    )
+
+    c_3 = Cylinder(
+        name="stationary_cylinder",
```
Comment (benflexcompute) [inline tests/simulation/params/test_validators_params.py:633]:
```suggestion
        name="another_cylinder",
```

**Review by benflexcompute — COMMENTED** [review]
Looking good. Can you fix the linter and formatting?

what I usually do is having these in my bashrc:
alias formatPy="/disk2/ben/testcases/src/Flow360Core/Flow360/.venv/bin/black ./ --exclude '/\.venv/' && /disk2/ben/testcases/src/Flow360Core/Flow360/.venv/bin/isort . --skip .venv" 
alias checkLint='/disk2/ben/testcases/src/Flow360Core/Flow360/.venv/bin/pylint $(git ls-files "flow360/*.py") --rcfile .pylintrc'
and replace "/disk2/ben/testcases/src/Flow360Core/Flow360/.venv/bin" with your own ones.

File: tests/simulation/params/test_validators_params.py (hunk 1 above)
Comment (anshulmittal-flex) [reply tests/simulation/params/test_validators_params.py]: Done. Thank you for your immense help.

**Review by NasserFlexCompute — COMMENTED** [review]
Let's avoid using abbreviations in the interface. How about calling the user entry "moving_reference_frame"? It'll be a lot easier for user to remember compared to isMRF.

File: flow360/component/simulation/models/volume_models.py [hunk 2]
```diff
@@ -436,9 +436,10 @@ class Rotation(Flow360BaseModel):
     )
     isMRF: Optional[bool] = pd.Field(
         None,
-        description="Flag to specify whether the MRF model is to be used for the rotation model. " + 
-        "Steady state simulation requires this flag to be True for all rotation models."
+        description="Flag to specify whether the MRF model is to be used for the rotation model. "
```
Comment (NasserFlexCompute) [inline flow360/component/simulation/models/volume_models.py:439]: "moving reference frame"

File: flow360/component/simulation/models/volume_models.py [hunk 3]
```diff
@@ -444,6 +444,12 @@ class Rotation(Flow360BaseModel):
         description="The parent rotating entity in a nested rotation case."
         + "The entity should be :class:`Cylinder` or :class:`GenericVolume` type.",
     )
+    noninertial_reference_frame_model: Optional[bool] = pd.Field(
+        None,
+        description="Flag to specify whether the noninertial referene frame model (MRF) is "
```
Comment (benflexcompute) [inline flow360/component/simulation/models/volume_models.py:449]: `noninertial referene frame model (MRF)` Abbreviation?

File: flow360/component/simulation/simulation_params.py [hunk 4]
```diff
@@ -316,6 +317,11 @@ def check_output_fields(params):
         """Only allow lowMachPreconditioner output field when the lowMachPreconditioner is enabled in the NS solver"""
         return _check_output_fields(params)
 
+    @pd.model_validator(mode="after")
+    def check_and_add_is_mrf_flag_in_volumezones(params):
```
Comment (benflexcompute) [inline flow360/component/simulation/simulation_params.py:321]:
```suggestion
    def check_and_add_noninertial_reference_frame_model_flag_in_volumezones(params):
```

Comment (benflexcompute) [reply flow360/component/simulation/simulation_params.py:321]: Also change the docstring please to be consistent.

File: flow360/component/simulation/models/volume_models.py (hunk 2 above)
Comment (anshulmittal-flex) [reply flow360/component/simulation/models/volume_models.py]: Done.

File: flow360/component/simulation/models/volume_models.py (hunk 3 above)
Comment (anshulmittal-flex) [reply flow360/component/simulation/models/volume_models.py:449]: Removed the abbreviation. I had left it their for ease of understanding to the clients. Thanks.

File: flow360/component/simulation/simulation_params.py (hunk 4 above)
Comment (anshulmittal-flex) [reply flow360/component/simulation/simulation_params.py:321]: Fixed. Thanks.

File: flow360/component/simulation/models/volume_models.py [hunk 5]
```diff
@@ -446,6 +446,12 @@ class Rotation(Flow360BaseModel):
         description="The parent rotating entity in a nested rotation case."
         + "The entity should be :class:`Cylinder` or :class:`GenericVolume` type.",
     )
+    noninertial_reference_frame_model: Optional[bool] = pd.Field(
```
[the remark this answers is not in the document]
Comment (anshulmittal-flex) [reply flow360/component/simulation/models/volume_models.py:449]: rotating_reference_frame could be confusing to the end user as they may think we are rotating the grid.
John and I discussed and John suggested using noninertial_reference_frame_model. This name states that we are using a model for the noninertial reference frame.

File: flow360/component/simulation/validation/validation_simulation_params.py [hunk 6]
```diff
@@ -341,3 +341,34 @@ def _check_parent_volume_is_rotating(models):
...
+
+        if model.noninertial_reference_frame_model is None:
+            model.noninertial_reference_frame_model = noninertial_reference_frame_default_value
+
+        if model.noninertial_reference_frame_model is False and is_steady is True:
+            raise ValueError(
+                f"For model #{model_index}, the noninertial_reference_frame_model is set to False but "
+                "the simulation is a steady state simulation. This is not allowed. All rotation models "
+                "should have this be true for a steady state simulation."
+            )
```
[the remark this answers is not in the document]
Comment (anshulmittal-flex) [reply flow360/component/simulation/validation/validation_simulation_params.py:372]: I have changed the message but left the name as noninertial_reference_frame_model.

**Review by benflexcompute — APPROVED** [review]

**Pull request merged**